tests/test_1d_upwind_laplace_matrix.cc

```diff
@@ -7,61 +7,130 @@
+
+   double mref_4th_order[] = {
+     1.336221213417364, -3.143071786321153,  2.411600780695188, -0.731187511242844,  0.111357511508477,  0.022476039032408, -0.007396247089440,                  0,                  0,                  0,
+     0.804447935880775, -1.360652110638654,  0.238604241481773,  0.398779563965364, -0.088162072611818,  0.002141425309071,  0.004841016613489,                  0,                  0,                  0,
+     0.267836979334519,  0.354996554399711, -1.677177847212427,  1.201904343321228, -0.110166180704016, -0.058673927711613,  0.021280078572598,                  0,                  0,                  0,
+     -0.210872649567740,  0.489776242991018,  0.992176068647725, -2.809296878224945,  1.720027308206487, -0.144379345689768, -0.057564974550696,  0.020134228187919,                  0,                  0,
+     0.037892486554968, -0.112039300610852, -0.094100279351347,  1.779750478630323, -3.229588496443332,  1.815279369236645, -0.162472035794183, -0.055555555555556,  0.020833333333333,                  0,
+     0.007648096615194,  0.002721394663611, -0.050117313253669, -0.149392517415107,  1.815279369236645, -3.254587948564049,  1.843959731543624, -0.215324384787472, -0.002703206562267,  0.002516778523490,
+     -0.002432322868339,  0.005945678122607,  0.017566776271339, -0.057564974550696, -0.157019954056124,  1.782081888203234, -3.104032161344934,  1.605461647766166, -0.045306699790849, -0.044699877752404,
+     0,                  0,                  0,  0.024390243902439, -0.065040650406504, -0.252087084629236,  1.944827524529746, -3.037407220285368,  1.427541994116328, -0.042224807227405,
+     0,                  0,                  0,                  0,  0.016393442622951, -0.002127113360472, -0.036889061578342,  0.959495438668352, -1.881782452274543,  0.944909745922054,
+     0,                  0,                  0,                  0,                  0,  0.007396247089440,  0.133963428792045,  0.553732231087227, -1.538939736908346,  0.843847829939636};
 
-template <typename OperatorType>
-bool test_matrix(OperatorType Dm)
-{
-  const int dim = 1;
-  const int n = 10;
-  unsigned int n_nodes[dim] = { n };
-  int n_nodes_tot = n_nodes[0];
-
-  typedef stenseal::CartesianGeometry<dim> Geometry;
-  Geometry geometry(n_nodes, dealii::Point<dim>(0.0),
-                    dealii::Point<dim>(1.0));
 
-  dealii::SparsityPattern sp_Laplace(n,n);
-  sp_Laplace.compress();
-  dealii::SparseMatrix<double> matrix_Laplace(sp_Laplace);
 
+template <typename OperatorType>
+     bool test_matrix(OperatorType Dm, int order)
+     {
+      const int dim = 1;
+      const int n = 10;
+      bool all_conv = false;
+      std::array<unsigned int,dim> n_nodes{ n };
+      int n_nodes_tot = n_nodes[0];
+
+
+      std::vector<dealii::Point<dim>> nodes(n_nodes_tot);
+      typedef stenseal::GeneralGeometry<dim> Geometry;
+      Geometry geometry(n_nodes,nodes);
```

Comment:
Detta borde inte funka, du får antingen använda `CartesianGeometry` som i t ex `test_1d_upwind_laplace`, eller om du vill köra allmän geometri får du skicka in en punkt-vektor som i `test_1d_general_upwind_laplace`. Right?